File: sdk/go-guest/templates/shipping.go
// lsf-example-begin: capsule
package export_examples_shipping_api

import wit "go.bytecodealliance.org/pkg/wit/types"

func Quote(items uint32, express bool) wit.Result[uint32, string] {
    if items < 1 || items > 100 { return wit.Err[uint32, string]("Choose between 1 and 100 items.") }
    base := uint32(500)
    if express { base = 1200 }
    return wit.Ok[uint32, string](base + items * 75)
}
// lsf-example-end: capsule

File: sdk/java-guest/templates/shipping.java
package dev.latent.app;
import dev.latent.generated.Bindings;
import dev.latent.guest.Result;

// lsf-example-begin: capsule
public final class Capsule implements Bindings.Exports {
    public Result<Long, String> quote(Long items, Boolean express) {
        if (items < 1 || items > 100) return Result.err("Choose between 1 and 100 items.");
        return Result.ok((express ? 1200L : 500L) + items * 75L);
    }
}
// lsf-example-end: capsule
